packages/core/data-transfer/lib/providers/local-file-destination-provider.ts

```diff
@@ -64,59 +63,63 @@ class LocalFileDestinationProvider implements IDestinationProvider {
       transforms.push(stringer());
     }
 
-    // Compression
-    if (this.options.compression.enabled) {
-      transforms.push(zip.createGzip());
-    }
-
-    // Encryption
-    if (this.options.encryption.enabled) {
-      if (!this.options.encryption.key) {
-        throw new Error("Can't encrypt without a key");
-      }
+    return transforms;
+  }
 
-      const cipher = createEncryptionCipher(this.options.encryption.key);
+  bootstrap(): void | Promise<void> {
+    const { compression, encryption } = this.options;
 
-      transforms.push(cipher);
+    if (encryption.enabled && !encryption.key) {
+      throw new Error("Can't encrypt without a key");
     }
 
-    return transforms;
-  }
+    this.#archive = tar.pack();
```

Comment:
Can we do this in the constructor and make #archive a required field so we don't have to do all the non-null assertions?

Reply:
We could but it kind of defeat the bootstrap method's role, isn't it?

Reply:
You're right, this way in theory we could initiate a second transfer and reset everything in bootstrap